apps/storage/routes/attachment/[orgSlug]/[attachmentId]/[filename].ts

```diff
@@ -0,0 +1,78 @@
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { GetObjectCommand } from '@aws-sdk/client-s3';
+import { s3Client } from '../../../../utils/s3';
+import { db } from '@uninbox/database';
+import { and, eq } from '@uninbox/database/orm';
+import { convoAttachments, orgMembers, orgs } from '@uninbox/database/schema';
+
+/**
+ * provides a proxy to attachments after verifying the user has access to the attachment
+ * path: storageServer/attachment/[orgSlug]/[attachmentId]/[filename.ext]
+ */
+export default eventHandler({
+  async handler(event) {
+    const orgSlug = getRouterParam(event, 'orgSlug');
+    const attachmentPublicId = getRouterParam(event, 'attachmentId');
+    const filename = getRouterParam(event, 'filename');
+
+    // attachment url: https://s3/attachments/[orgPublicId]/[attachmentId]/[filename].ext
+
+    const attachmentQueryResponse = await db.query.convoAttachments.findFirst({
+      where: eq(convoAttachments.publicId, attachmentPublicId),
+      columns: {
+        fileName: true,
+        orgId: true,
+        public: true
+      }
+    });
+    if (
+      !attachmentQueryResponse ||
+      attachmentQueryResponse.fileName !== filename
+    ) {
+      return `Attachment ${filename} not found`;
+    }
+
+    const orgQueryResponse = await db.query.orgs.findFirst({
+      where: eq(orgs.slug, orgSlug),
+      columns: {
+        id: true,
+        publicId: true
+      }
+    });
+    if (!orgQueryResponse) {
+      setResponseStatus(event, 400);
+      return send(event, 'Invalid org');
+    }
+
+    if (!attachmentQueryResponse.public) {
+      const userId = +event.context.user.id;
```

Comment:
Do we need the `+` here?

Reply:
issue with returns sometimes being strings